### ISSUE TITLE
Validate relation names in `with()`, `load()`, `has()`, `whereHas()`

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#643]
## Problem

Passing a non-existent relation name to `with()`, `load()`, `has()`, `whereHas()`, etc. is silently accepted by the plugin. This is one of the most common sources of runtime errors in Laravel apps.

```php
// Should report: relation 'nonExistent' not found on User
User::with('nonExistent')->get();
User::has('typoRelation')->get();
$user->load('missingRelation');
```

## Scope

Validate relation name strings passed to:
- `Builder::with()`, `Builder::without()`
- `Builder::has()`, `Builder::orHas()`, `Builder::doesntHave()`, `Builder::orDoesntHave()`
- `Builder::whereHas()`, `Builder::orWhereHas()`, `Builder::whereDoesntHave()`, `Builder::orWhereDoesntHave()`
- `Builder::withWhereHas()`
- `Builder::whereRelation()`, `Builder::orWhereRelation()`, `Builder::whereMorphRelation()`
- `Model::load()`, `Model::loadMissing()`, `Model::loadCount()`, `Model::loadSum()`, etc.

### Features
- **Dot-notation nested relations**: `with('posts.comments.author')` — walk each segment and resolve the intermediate model
- **Array syntax**: `with(['posts', 'comments'])` — check each element
- **Select syntax**: `with('posts:id,title')` — strip the `:columns` part before checking
- **Closure syntax**: `with(['posts' => fn($q) => ...])` — validate the array key

## Implementation

`AfterMethodCallAnalysisInterface` handler. Resolve the model type from `Builder<TModel>` or `Model` generic param, then check that each relation name corresponds to a method returning a `Relation` subtype.

## Reference

Larastan implements this via `RelationExistenceRule` — see `.alies/larastan/src/Rules/RelationExistenceRule.php`.
